refactor: extract test

coderbyheart opened the pull request:
Make the test set-up isolated and do not use magic
to construct the seed data.

This makes the seeded data explicit, but still
random.

coderbyheart committed:
refactor: extract test

Make the test set-up isolated and do not use magic
to construct the seed data.

This makes the seeded data explicit, but still
random.

coderbyheart commented on e2e-tests/e2eTest.spec.ts:
```diff
@@ -65,65 +67,29 @@ void describe('e2e-tests', async () => {
-				lastWeek: 138,
-			},
-			ts: now.toISOString(),
-			usedBytes: 0,
```
@Lenakh97 why is this `0`? Shouldn't that also be `138`?

Lenakh97 replied:
Correct. DynamoDB and Timestream works independently, and I'm seeding DynamoDB with 0 here. So that's why it's 0. But it should be 138 or a higher number.

coderbyheart replied:
It's bad to assert for something that we know is wrong. Instead, only assert what should work.

I've split up this test in two, so one focuses on the base feature (get SIM data), and then the other one on another feature, usage per timespan.

https://github.com/bifravst/sim-details/pull/193/commits/c39e09195667d170bca365fa25f7ee3a1f3b2f55

coderbyheart replied:
In https://github.com/bifravst/sim-details/pull/193/commits/639d7f53040b2abd7c13f8ae6939659a5c52c273 I've further separate this and moved the seeders to a separate file.

coderbyheart committed:
ci: this is not applicable in this project

coderbyheart committed:
test(e2e): run all e2e tests

coderbyheart committed:
test(e2e): separate tests

It is better to not test for incorrect data,
instead only test what should work.

coderbyheart committed:
refactor(e2e): move shared clients to function

coderbyheart committed:
test(e2e): mark as todo

This no longer works because it depended on the first
test to run

Lenakh97 committed:
refactor(e2e): extract test cases to individual files

coderbyheart commented on lambda/getBasicInformationLambda.ts:
```diff
@@ -44,7 +52,16 @@ const validIssuers: Record<string, string> = {
 const listRecordsForIntervalFunc = listRecordsForInterval(ts, dbName, tableName)
 const { track, metrics } = metricsForComponent('getAllSimUsageOnomondo')
-
+// Do not cache the result if we are in test mode
+const availableColumnsCache =
+	isTest === '1'
+		? async () => {
+				console.warn(`Fetching available columns for ${dbName}.${tableName}.`)
+				const cols = await getAvailableColumns(ts, dbName, tableName)()
+				console.warn(`Available columns`, JSON.stringify(cols))
+				return cols
+			}
+		: once(getAvailableColumns(ts, dbName, tableName))
```
*problem:* new behaviour, not a refactoring

coderbyheart commented on cdk/BackendStack.ts:
```diff
@@ -131,6 +131,7 @@ export class BackendStack extends Stack {
 					SIM_DETAILS_JOBS_QUEUE: resolutionJobsQueue.queueUrl,
 					WIRELESS_LOGIC_QUEUE: wirelessLogicQueue.queueUrl,
 					TABLE_INFO: table.ref,
+					IS_TEST: isTest ? '1' : '0',
```
*problem:* new behaviour, not a refactoring

coderbyheart commented on cdk/layer.ts:
```diff
@@ -11,6 +11,7 @@ const dependencies: Array<keyof (typeof pJson)['dependencies']> = [
 	'@bifravst/timestream-helpers',
 	'@middy/core',
 	'@aws-lambda-powertools/metrics',
+	'lodash-es',
```
*problem:* new behaviour, not a refactoring

coderbyheart commented:
Otherwise looks good.
